src/target/cortexm.c

```diff
@@ -227,6 +227,27 @@ static void cortexm_mem_write(target_s *target, target_addr64_t dest, const void
 	adiv5_mem_write(cortex_ap(target), dest, src, len);
 }
 
+bool target_is_cortexm(target_s *target)
+{
+	if ((target == NULL) || (target->priv_free != cortex_priv_free)) {
```

Comment:
Please drop the extra parens, they suppress the GCC warning about accidental assignment which we want here. If the condition being checked is inverter using De Morgans' law, it can be turned into a direct return as the boolean checks already convert the result to bool. Eg:

Reply:
Good suggestion. I've updated that.